koa_crud/test/integration/models/promo.spec.ts

```diff
@@ -0,0 +1,165 @@
+      });
+
+      it('should throw an error for empty template', async () => {
+        const data = {
+          name: chance.name(),
+          template: '',
+          title: chance.word(),
+          description: chance.sentence(),
+          minimumBalance: chance.prime(),
+        };
+
+        await expect(PromoModel.create(data)).to.eventually.rejected;
+      });
+
+      it('should throw an error for invalid template', async () => {
+        const data = {
+          name: chance.name(),
+          template: 'Test',
+          title: chance.word(),
+          description: chance.sentence(),
+          minimumBalance: chance.prime(),
+        };
+
+        await expect(PromoModel.create(data)).to.eventually.rejected;
+      });
+
+      it('should throw an error for empty title', async () => {
+        const data = {
+          name: chance.name(),
+          template: PromoTemplate.Deposit,
+          title: '',
+          description: chance.sentence(),
+          minimumBalance: chance.prime(),
+        };
+
+        await expect(PromoModel.create(data)).to.eventually.rejected;
+      });
+
+      it('should throw an error for empty description', async () => {
+        const data = {
+          name: chance.name(),
+          template: PromoTemplate.Deposit,
+          title: chance.word(),
+          description: '',
+          minimumBalance: chance.prime(),
+        };
+
+        await expect(PromoModel.create(data)).to.eventually.rejected;
+      });
+
+      it('should throw an error for invalid status', async () => {
+        const data = {
+          name: chance.name(),
+          template: PromoTemplate.Deposit,
+          title: chance.word(),
+          description: chance.sentence(),
+          minimumBalance: chance.prime(),
+          status: 'TEST',
+        };
+
+        await expect(PromoModel.create(data)).to.eventually.rejected;
+      });
+    });
+
+    context('Given correct values', () => {
+      it('should insert promo', async () => {
+        const data = {
+          name: chance.name(),
+          template: PromoTemplate.Deposit,
+          title: chance.word(),
+          description: chance.sentence(),
+          minimumBalance: chance.prime(),
+        };
+
+        const result = await PromoModel.create(data);
+
+        expect(result).to.be.an('object');
+        expect(result.name).to.equal(data.name);
+      });
+    });
+  });
+
+  describe('List Promos', () => {
+    it('should retrieve all promo', async () => {
+      const result = await PromoModel.find({}).lean();
+
+      expect(result).to.be.an('array');
+      expect(result).to.have.length.greaterThan(0);
+    });
+
+    it('should retrieve one promo', async () => {
+      const promo = await PromoModel.findOne({}).lean({ virtuals: true });
+
+      const result = await PromoModel.findById(promo.id);
+
+      expect(result).to.be.an('object');
+      expect(result.id).to.equal(promo.id);
+    });
+  });
```

Comment:
This should not be tested. It is not worth it. Since the `Model.findById` and `Model.find` is already tested in mongoose library.

I think the most noteworthy when testing the Model is the default value of fields when you create something.